src/test/java/co/rsk/federate/config/PowHSMBookkeepingConfigTest.java

```diff
@@ -1,57 +1,145 @@
-        powHsmBookkeepingConfig = new PowHSMBookkeepingConfig(signerConfigMock, NetworkParameters.ID_REGTEST);
-        assertEquals(PowHSMBookkeepingConfig.DIFFICULTY_CAP_REGTEST, powHsmBookkeepingConfig.getDifficultyCap());
+    void getMaxChunkSizeToHsm_whenCustomConfigNotAvailable_shouldReturnDefaultConfig() {
+        when(config.hasPath("bookkeeping.maxChunkSizeToHsm")).thenReturn(false);
+
+        assertEquals(DEFAULT_MAX_CHUNK_SIZE, powHsmBookkeepingConfig.getMaxChunkSizeToHsm());
     }
 
     @Test
-    void testGetDifficultyCapForInvalidNetwork() {
-        when(signerConfigMock.getConfig()).thenReturn(configMock);
-        assertThrows(IllegalArgumentException.class, () -> new PowHSMBookkeepingConfig(signerConfigMock, NetworkParameters.ID_UNITTESTNET));
+    void getInformerInterval_whenCustomConfigAvailable_shouldReturnCustomConfig() {
+        long customInformerInterval = 1L; 
+        when(config.hasPath("bookkeeping.informerInterval")).thenReturn(true);
+        when(config.getLong("bookkeeping.informerInterval")).thenReturn(customInformerInterval);
+
+        assertEquals(customInformerInterval, powHsmBookkeepingConfig.getInformerInterval());
+    }
+
+    @Test
+    void getInformerInterval_whenCustomConfigNotAvailable_shouldReturnDefaultConfig() {
+        when(config.hasPath("bookkeeping.informerInterval")).thenReturn(false);
+
+        assertEquals(DEFAULT_INFORMER_INTERVAL, powHsmBookkeepingConfig.getInformerInterval());
+    }
+
+    @Test
+    void isStoppingBookkepingScheduler_whenCustomConfigAvailable_shouldReturnCustomConfig() {
+        boolean customIsStoppingInformerInterval = !DEFAULT_STOP_BOOKKEPING_SCHEDULER; 
+        when(config.hasPath("bookkeeping.stopBookkeepingScheduler")).thenReturn(true);
+        when(config.getBoolean("bookkeeping.stopBookkeepingScheduler")).thenReturn(customIsStoppingInformerInterval);
+
+        assertTrue(powHsmBookkeepingConfig.isStopBookkeepingScheduler());
+    }
+
+    @Test
+    void isStoppingBookkepingScheduler_whenCustomConfigNotAvailable_shouldReturnDefaultConfig() {
+        when(config.hasPath("bookkeeping.stopBookkeepingScheduler")).thenReturn(false);
+
+        assertFalse(powHsmBookkeepingConfig.isStopBookkeepingScheduler());
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideNetworkParametersAndExpectedCaps")
+    void getDifficultyCap_whenGivenNetwork_shouldMatchExpectedDifficultyCap(String networkId,
+          NetworkDifficultyCap expectedNetworkDifficultyCap) {
+        PowHSMBookkeepingConfig powHsmBookkeepingConfig = new PowHSMBookkeepingConfig(signerConfig, networkId);
+
+        assertEquals(expectedNetworkDifficultyCap.getDifficultyCap(),
+            powHsmBookkeepingConfig.getDifficultyCap());
+    }
+
+    @Test
+    void getDifficultyCap_whenGivenInvalidNetwork_showThrowIllegalArgumentException() {
+        PowHSMBookkeepingConfig powHsmBookkeepingConfig =
+            new PowHSMBookkeepingConfig(signerConfig, NetworkParameters.ID_UNITTESTNET);
```

Comment:
i'd call it `UNKNOWN_NETWORK` or similar, it was a bit confusing for me at first

Reply:
I agree, but that class is out of the scope of this change :)

Reply:
haha you right, my bad